File: src/tools/hash/genhash.py
import hashlib

def hash_string(string, algorithm):
    if algorithm == 'md5':
        return hashlib.md5(string.encode()).hexdigest()
    elif algorithm == 'sha1':
        return hashlib.sha1(string.encode()).hexdigest()
    elif algorithm == 'sha256':
        return hashlib.sha256(string.encode()).hexdigest()
    else:
        return "Error: Unsupported algorithm"
    
if __name__ == "__main__":
    string = input("Enter string: ")
    algorithm = input("Enter algorithm: ")
    print(hash_string(string, algorithm))

File: src/tools/hash/genhash.go
package main

import (
	"crypto/md5"
	"crypto/sha1"
	"crypto/sha256"
	"fmt"
	"os"
)

func hashString(s string, algorithm string) string {
	switch algorithm {
	case "md5":
		h := md5.New()
		h.Write([]byte(s))
		return fmt.Sprintf("%x", h.Sum(nil))
	case "sha1":
		h := sha1.New()
		h.Write([]byte(s))
		return fmt.Sprintf("%x", h.Sum(nil))
	case "sha256":
		h := sha256.New()
		h.Write([]byte(s))
		return fmt.Sprintf("%x", h.Sum(nil))
	default:
		return "Error: Unsupported algorithm"
	}
}

func main() {
	if len(os.Args) != 3 {
		return
	}

	string := os.Args[1]
	algorithm := os.Args[2]

	fmt.Println(hashString(string, algorithm))
}
